components/canvas/webgl_thread.rs

```diff
@@ -277,18 +312,33 @@ impl<VR: WebVRRenderHandler + 'static> WebGLThread<VR> {
             .gl_factory
             .new_shared_context(version, size, attributes)
             .map(|r| (r, WebGLContextShareMode::SharedTexture))
+            .or_else(|err| {
+                #[cfg(target_os = "macos")]
```

Comment:
Should we introduce a new abstraction so we're not special-casing `macos`? Something where the GL context isn't sharing, but the texture is? Alternatively, should we move this logic into `GLFactory`?